File: Ravi-Teja/SortArrayByParity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        int i=0, j=nums.size()-1;
        while(i<j){
            while(i<j && nums[i]%2==0){
                i++;
            }
            while(i<j && nums[j]%2!=0){
                j--;
            }
            if(i<j){
                swap(nums[i], nums[j]);    
                i++;
                j--;
            }
        }
        return nums;
    }
};

File: arun_lodhi/Sort_Array_By_Parity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        l = []
        l2 = []
        for x  in nums:
            if x%2==0:
                l.append(x)
            else:
                l2.append(x)
        l.sort()
        l.reverse()
        l2.sort()
        l.extend(l2)
        return l
        
